packages/overlay/src/integrations/sentry/store/slices/eventsSlice.ts

```diff
@@ -28,22 +29,48 @@ export const createEventsSlice: StateCreator<SentryStore, [], [], EventsSliceSta
       event.event_id = generateUuidv4();
     }
 
-    const { eventIds, events } = get();
-    if (eventIds.has(event.event_id)) return;
+    const { eventsById, events } = get();
+    if (eventsById.has(event.event_id)) return;
 
-    const newEventIds = new Set(eventIds);
-    newEventIds.add(event.event_id);
-    set({ eventIds: newEventIds });
+    const newEventIds = new Map(eventsById);
+    newEventIds.set(event.event_id, event);
+    set({ eventsById: newEventIds });
 
     if (isErrorEvent(event)) {
       await get().processStacktrace(event);
     }
 
-    event.timestamp = toTimestamp(event.timestamp);
+    if (event.timestamp) {
+      event.timestamp = toTimestamp(event.timestamp);
+    }
     if (event.start_timestamp) {
       event.start_timestamp = toTimestamp(event.start_timestamp);
     }
 
+    if (isLogEvent(event) && event.items?.length) {
+      const { logsById, logsByTraceId } = get();
+      for (const logItem of event.items) {
+        if (logItem.severity_number == null) {
+          logItem.severity_number = 0;
+        }
+        logItem.sdk = logItem.attributes?.['sentry.sdk.name']?.value as string;
+        logItem.timestamp = toTimestamp(logItem.timestamp);
+        logItem.id = generateUuidv4();
```

Comment:
weren't we going to add a prefix to ids?

Reply:
Yes but in a follow up :)